docs/features/type-checking.md

```diff
@@ -0,0 +1,223 @@
+# Type checking features
+
+You can generally expect ty to support all typing features that are described and specified in the
+[Python typing documentation]. This page highlights some of the unique features that ty's type
+system provides.
+
+## Redeclarations
+
+ty allows you to reuse the same symbol with a different type. The following example shows how the
+`paths` parameter is redeclared as a list of strings:
+
+```py
+def split_paths(paths: str) -> list[Path]:
+    paths: list[str] = paths.split(":")
+    return [Path(p) for p in paths]
+```
+
+(Full example in the [playground](https://play.ty.dev/80a74c95-a43e-4a3d-8c26-f88e879d7dcb))
+
+## Intersection types
+
+ty has first-class support for intersection types. In contrast to a union type `A | B`, which means
+"*either* A *or* B", an intersection type `A & B` means "*both* A *and* B". Type narrowing in ty is
+based on intersections. For example, notice how we can call `obj.serialize_json()` *and* access the
+`.version` property in the following function:
+
+```py
+def output_as_json(obj: Serializable) -> str:
+    if isinstance(obj, Versioned):
+        reveal_type(obj)  # reveals: Serializable & Versioned
+
+        return str({
+            "data": obj.serialize_json(),
+            "version": obj.version
+        })
+    else:
+        return obj.serialize_json()
+```
+
+(Full example in the [playground](https://play.ty.dev/39241435-5e78-4ce9-817f-ce65be73a6ed))
```

Comment:
I'm not sure if this implies any change to this section, but it's worth noting that all major type checkers have enough "implicit synthesized intersection" support that they support this example as well as we do. So this isn't a unique ty feature.